### PR TITLE
fix: [CDS-81701]: fixed to remove the extra entry on spec for ondelegate

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Problem: 
 when onDelegate field is configured , yaml is not configured right, there is an extra entry for onDelegate

Solution:
   Removed and fixed the logic for yaml formatting when onDelegate is configured

### CODE REVIEW

**Comment by AnshuH** [comment]
trigger recency

**Comment by sahithikolichala** [comment]
retrigger messageCheck

**Comment by sahithikolichala** [comment]
retrigger messageCheck

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ExecutionTargetGroup.tsx [hunk 1]
```diff
@@ -94,11 +89,11 @@ export function MultiTypeExecutionTargetGroup(props: ConnecteExecutionTargetGrpP
         >
           <RadioGroup
             data-tooltip-id="executionTargetTooltip"
-            selectedValue={formik.values.spec?.onDelegate}
+            selectedValue={selectedValue}
             disabled={readonly}
             inline={true}
             onChange={e => {
-              formik.setFieldValue('spec.onDelegate', e.currentTarget.value)
+              formik.setFieldValue(name, e.currentTarget.value === 'targethost' ? false : true)
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ExecutionTargetGroup.tsx:96]:
```suggestion
              formik.setFieldValue(name, e.currentTarget.value !== 'targethost')
```

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx [hunk 2]
```diff
@@ -70,8 +71,10 @@ export const FixedExecTargetGroup = ({
   const { getString } = useStrings()
   const formValues = formik.values
   const { setFieldValue } = formik
+  const fieldName = prefix ? `${prefix}spec.onDelegate` : 'spec.onDelegate'
+  const onDelegateVal = get(formValues, `${fieldName}`)
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/OptionalConfiguration.tsx:75]:
```suggestion
  const onDelegateVal = get(formValues, fieldName)
```

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptInputSetStep.tsx [hunk 3]
```diff
@@ -158,6 +157,23 @@ export default function ShellScriptInputSetStep(props: ShellScriptInputSetStepPr
           </MultiTypeFieldSelector>
         </div>
       ) : null}
+
+      {getMultiTypeFromValue(template?.spec?.onDelegate) === MultiTypeInputType.RUNTIME && (
+        <>
+          <Label>
+            <HarnessDocTooltip tooltipId={'exec-target'} labelText={'Execution Target'} />
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/ShellScriptInputSetStep.tsx:164]: Can we add the label to strings?

File: src/modules/75-cd/components/PipelineSteps/ShellScriptStep/helper.ts [hunk 4]
```diff
@@ -1,14 +1,9 @@
...
 
 import { ShellScriptData, ShellScriptFormData } from './shellScriptTypes'
 
 const getOnDelegateValue = (values: ShellScriptData): string => {
-  if (getMultiTypeFromValue(values?.spec?.onDelegate) === MultiTypeInputType.FIXED) {
-    return values.spec?.onDelegate ? 'delegate' : 'targethost'
-  } else {
-    return values.spec?.onDelegate
-  }
+  return values.spec?.onDelegate !== '' ? values.spec?.onDelegate : true
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/ShellScriptStep/helper.ts:6]: The function's return type is `string`, but this ternary can return true.

**Review by rameshx — CHANGES_REQUESTED** [review]

**Comment by sahithikolichala** [comment]
retrigger typecheck

**Review by rameshx — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**